remove peerDependencies

A missing `peerDependency` will be reduced to a warning with npm v3.0.0, which is now in beta.

This package requires `istanbul`, among other things, to be installed.

It just so happens that a *fork* of `istanbul` (`istanbul-harmony`) works better than the `harmony` branch of `istanbul` itself.  

The `istanbul-harmony` package uses the same executable, `node_modules/.bin/istanbul`.  This means they cannot coexist; if `istanbul-harmony` is installed, and `grunt-mocha-istanbul` is installed thereafter, the presence of the peer dependency creates a conflict, and npm will fail (let me know if you'd like me to paste logs).

This is a non-issue in npm v3.0.0, because it does not attempt to install this package's `peerDependencies`.

`README.md` mentions that you may use something else like `ibrik` if you wish, but `ibrik` does not use the same executable.  Using `ibrik` doesn't stop this package from installing `istanbul` anyway (and checking for its presence), even if it will remain unused--which this package should stop doing anyway.

If the intent is to let the user use whatever packages (or forks!) they wish, then:

- **the `peerDependencies` property of `package.json` should be removed**
- **explicit checks in `index.js` for the dependencies should be removed**
- **at execution time, catch errors in the case of a non-existent `mocha`, `grunt`, or `istanbul`, and display a friendly warning message**
- **update `README.md` to instruct users to explictly install the three dependencies**

I can provide a PR if you wish.

@voxpelli Has a reasonable argument in the case of `grunt`, but the fact of the matter is that if I have both `grunt-mocha-istanbul` and `istanbul-harmony` in my `devDependencies`, `npm install` *will fail* in npm <v3.

```
npm ERR! Linux 3.13.0-40-generic
npm ERR! argv "/home/travis/.nvm/versions/io.js/v2.4.0/bin/iojs" "/home/travis/.nvm/versions/io.js/v2.4.0/bin/npm" "install"
npm ERR! node v2.4.0
npm ERR! npm  v2.13.0
npm ERR! path ../istanbul/lib/cli.js
npm ERR! code EEXIST
npm ERR! errno -17
npm ERR! syscall symlink
npm ERR! EEXIST: file already exists, symlink '../istanbul/lib/cli.js' -> '/home/travis/build/digsjs/digs-local/node_modules/.bin/istanbul'
File exists: ../istanbul/lib/cli.js
Move it away, and try again. 
npm ERR! Please include the following file with any support request:
npm ERR!     /home/travis/build/digsjs/digs-local/npm-debug.log
```

Part of the selling point of this package is the `scriptPath` option, which should allow me to use whatever `istanbul`-compatible coverage tool I want.  If `istanbul` as a `peerDependency` implies `istanbul` must be installed, which is plainly not true.

I could go either way with `mocha` being a `peerDependency`.  If users were allowed to specify Mocha's executable in addition to Istanbul's, then it shouldn't be.

I'm happy to restore `grunt` as a `peerDependency` in #40.  This is, after all, a `grunt` plugin.  It's not a `mocha` plugin nor is it an `istanbul` plugin...

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -29,7 +29,7 @@ module.exports = function (grunt){
     done();
   });
 
-  grunt.loadNpmTasks('grunt-mocha-istanbul');
+  require('./tasks')(grunt);
 
   grunt.registerTask('default', ['mocha_istanbul']);
 };

--- a/README.md
+++ b/README.md
@@ -5,15 +5,25 @@
 grunt mocha istanbul task
 ==============
 
-[Mocha](http://visionmedia.github.com/mocha/) reporter to generate coverage report of [istanbul](http://gotwarlost.github.com/istanbul/) instrumented code, for grunt
+[Mocha](https://mochajs.org) reporter to generate coverage report of [istanbul](http://gotwarlost.github.com/istanbul/) instrumented code, for grunt
 This doesn't force you to use PhantomJS, or instrument code for server or client-side.
 
 Install
 ==============
 
-1. Install it using `npm install grunt-mocha-istanbul --save-dev`
-2. It needs `mocha`, `grunt` and `istanbul` to be installed locally on your project (aka, having them in your devDependencies)
-3. Call inside Gruntfile.js `grunt.loadNpmTasks('grunt-mocha-istanbul')`
+1. Install needed dependencies using: `npm install grunt mocha istanbul --save-dev`
+2. Install this package using: `npm install grunt-mocha-istanbul --save-dev`
+3. Call inside `Gruntfile.js`: `grunt.loadNpmTasks('grunt-mocha-istanbul')`
+
+Changes from 2.x
+==============
+
+Peer dependencies for `mocha` and `istanbul` have been removed.  You should `npm install` the following modules yourself:
+  
+- `mocha`
+- `istanbul` (or a compatible module; see `scriptPath` usage below)
+
+If using `npm` version < 3, you will need to install `grunt` as well.  
 
 Changes from 1.x
 ==============

--- a/package.json
+++ b/package.json
@@ -3,6 +3,9 @@
   "version"         : "2.4.0",
   "description"     : "Almost config-free Istanbul code coverage reporter for Mocha usage in Grunt",
   "peerDependencies": {
+    "grunt": "0.4.x"
+  },
+  "devDependencies": {
     "grunt": "0.4.x",
     "mocha": ">=1.x.x",
     "istanbul": "0.x.x"
@@ -10,6 +13,9 @@
   "repository"      : {
     "type": "git",
     "url" : "git://github.com/pocesar/grunt-mocha-istanbul.git"
+  },
+  "scripts": {
+    "test": "grunt"
   },
   "keywords"        : [
     "grunt",

--- a/tasks/index.js
+++ b/tasks/index.js
@@ -3,22 +3,22 @@ module.exports = function (grunt){
 
   var
     cmd = process.execPath,
-    path = require('path'),
-    mochaPath,
-    istanbulPath;
+    path = require('path');
 
-  try {
-    mochaPath = require.resolve('mocha/bin/_mocha');
-  } catch (e) {
-    grunt.log.error('Missing mocha peer dependency');
-    return;
+  function getMochaPath() {
+    try {
+      return require.resolve('mocha/bin/_mocha');
+    } catch (ignored) {
+      grunt.fail.warn('Mocha peer dependency missing.  Please "npm install mocha"');
+    }
   }
 
-  try {
-    istanbulPath = require.resolve('istanbul/lib/cli');
-  } catch (e) {
-    grunt.log.error('Missing istanbul peer dependency');
-    return;
+  function getIstanbulPath() {
+    try {
+      return require.resolve('istanbul/lib/cli');
+    } catch (ignored) {
+      grunt.fail.warn('Istanbul peer dependency missing.  Please "npm install istanbul"');
+    }
   }
 
   function executeCheck(callback, coverageFolder, options) {
@@ -84,7 +84,6 @@ module.exports = function (grunt){
     var
       done = this.async(),
       options = this.options({
-        scriptPath: istanbulPath,
         coverageFolder: 'coverage',
         check: {
           statements: false,
@@ -93,6 +92,11 @@ module.exports = function (grunt){
           branches: false
         }
       });
+
+    // only execute this function if no scriptPath is specified
+    if (!options.scriptPath) {
+      options.scriptPath = getIstanbulPath();
+    }
 
     executeCheck(function (err, result) {
       if (err) { return done(err); }
@@ -115,6 +119,7 @@ module.exports = function (grunt){
     }
 
     var
+      mochaPath = getMochaPath(),
       options = this.options({
         require        : [],
         ui             : false,
@@ -142,14 +147,17 @@ module.exports = function (grunt){
         },
         excludes       : false,
         mochaOptions   : false,
-        istanbulOptions: false,
-        scriptPath     : istanbulPath
+        istanbulOptions: false
       }),
       filesDir = grunt.file.isDir(this.filesSrc[0]) ? this.filesSrc[0] : path.dirname(this.filesSrc[0]),
       coverageFolder = path.join(process.cwd(), options.coverageFolder),
       rootFolderForCoverage = options.root ? path.join(process.cwd(), options.root) : '.',
       done = this.async(),
       args = [];
+
+    if (!options.scriptPath) {
+      options.scriptPath = getIstanbulPath();
+    }
 
     if (options.harmony) {
       args.push('--harmony');

